packages/compass-context-menu/src/context-menu-provider.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+  createContext,
+} from 'react';
+import type { ContextMenuContext, ContextMenuState } from './types';
+import type { EnhancedMouseEvent } from './context-menu-content';
+import { getContextMenuContent } from './context-menu-content';
+
+export const Context = createContext<ContextMenuContext | null>(null);
+
+export function ContextMenuProvider({
+  children,
+  wrapper,
+}: {
+  children: React.ReactNode;
+  wrapper: React.ComponentType<{
```

Comment:
I saw you mentioned this in the follow up pr: https://github.com/mongodb-js/compass/pull/6956/files#r2104832123 
`wrapper` here could be a bit ambiguous, although we're only really consuming this in compass-components in one place, maybe we rename it to something a bit more descriptive like `menuWrapper`?
(Not a blocker, also feel free to make any changes we might want in that follow up)